File: frontend/src/microservices/flight/objects/FlightAirport.ts
// 本文件定义 flight 模块的 `FlightAirport`，作为传输数据并提供 JSON 编解码。

export type FlightAirport = {
  airportCode: string
  cityName: string
  airportName: string
}

export const flightAirportFromJson = (json: string): FlightAirport =>
  JSON.parse(json) as FlightAirport

export const flightAirportToJson = (value: FlightAirport): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/flight/objects/FlightAirport.scala
// FlightAirport is a backend-only nested value object for airport code, city, and airport name.
// The planner uses it when shaping grouped airport data for suggestions and flight responses, but the browser only sees the final planner output.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class FlightAirport(
    airportCode: String,
    cityName: String,
    airportName: String
)

object FlightAirport:
  given sourceEncoder: Encoder[FlightAirport] = deriveEncoder
  given sourceDecoder: Decoder[FlightAirport] = deriveDecoder

